File: Java/src/main/java/QuestionBank/Solution001_100/Solution88.java
package QuestionBank.Solution001_100;

/**
 * @author windows
 */
public class Solution88 {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        while(m > 0 || n > 0){
            int a = m > 0 ? nums1[m - 1] : Integer.MIN_VALUE;
            int b = n > 0 ? nums2[n - 1] : Integer.MIN_VALUE;
            if (a >= b){
                nums1[m + n - 1] = a;
                m--;
            }else{
                nums1[m + n - 1] = b;
                n--;
            }
        }
    }

    public static void main(String[] args) {
        Solution88 solution88 = new Solution88();
        solution88.merge(new int[]{1,2,3,0,0,0}, 3, new int[]{2, 5, 6}, 3);
    }



}

File: C++/src/Solution1_100/Solution88.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution88.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

void Solution88::merge(vector<int> &nums1, int m, vector<int> &nums2, int n) {
    int i = m - 1;
    int j = n - 1;
    int k = m + n - 1;
    while (i >= 0 && j >= 0){
        if (nums1[i] > nums2[j]){
            nums1[k--] = nums1[i--];
        } else{
            nums1[k--] = nums2[j--];
        }
    }
    while (j >= 0){
        nums1[k--] = nums2[j--];
    }
}
